app/models/service_retire_task.rb

```diff
@@ -14,5 +14,49 @@ def update_and_notify_parent(*args)
   end
 
   def task_finished
+    update_attributes(:status => status == 'Ok' ? 'Completed' : 'Failed')
+  end
+
+  def task_active
+    update_attributes(:status => 'Active')
+  end
+
+  def after_request_task_create
+    update_attributes(:description => get_description)
+    parent_svc = Service.find_by(:id => options[:src_id])
+    _log.info("- creating service tasks for service <#{self.class.name}:#{id}>")
+
+    create_retire_subtasks(parent_svc)
+  end
+
+  def create_retire_subtasks(parent_service)
+    parent_service.service_resources.collect do |svc_rsc|
+      next unless svc_rsc.resource.present? && svc_rsc.resource.respond_to?(:retire_now)
+      next if svc_rsc.resource_type == "ServiceTemplate"
+
+      nh = parent_service_task.attributes.dup
+      %w(id created_on updated_on type state status message).each { |key| nh.delete(key) }
+      nh['options'] = parent_service_task.options.dup
+      nh['options'].delete(:child_tasks)
+      # Initial Options[:dialog] to an empty hash so we do not pass down dialog values to child services tasks
+      nh['options'][:dialog] = {}
+      next if child_svc_rsc.resource_type == "ServiceTemplate" &&
+              !self.class.include_service_template?(parent_service_task,
+                                                    child_svc_rsc.resource.id,
+                                                    parent_service)
+      new_task = (svc_rsc.resource.type.demodulize + "RetireTask").constantize.new(nh)
+      new_task.options.merge!(
```

Comment:
Can't this live in the `new` call?  Is options initialized as a non-empty hash?

Reply:
Unfortunately, it is, yes.